mercantile/__init__.py

```diff
@@ -618,7 +632,7 @@ def merge(merge_set):
     for tile in sorted(tiles, key=operator.itemgetter(2)):
         x, y, z = tile
         is_new_tile = True
-        for supertile in (parent(tile, zoom=i) for i in range(z + 1)):
+        for supertile in (parent(tile, zoom=i) for i in range(z)):
```

Comment:
This was uncovered by adding the pre-condition checks: parent(t, zoom=t.z) makes semantically no sense, see above